### ISSUE TITLE
Page error 0xfffffffffff8 in layer primary2 (Page Fault at entry 0xc30063 in table page table)

### ISSUE THREAD

**Issue by jackdyson1399** [volatilityfoundation/volatility3#356]
Running Volatility3 through RAM Capture from a Windows10 19041 VM the above error is returned. Attempted on volatility2 but the address space for that version of windows cannot be found and produces failed results too.
I have tried this on five different RAM Captures taken from the same system with different RAM capture software, so doubt it to be a capture issue as error persists with all dumps:
- DumpIT
- Belkasoft RAM Capture
- Magnet RAM Capturer (both full and segmented)
- FTK Imager 

![image](https://user-images.githubusercontent.com/46749580/95740532-9d58f000-0c84-11eb-9a69-6085b7d5b902.png)

**Comment by atcuno** [comment]
1) What is the VM software that you are using? If its VMware fusion or Workstation, can you suspend the VM and then run the resulting *.vmem through Volatility 2 and 3? If you move or rename the *.vmem, then need to do the same steps for the associated *.vmss or analysis will not work. None of those capture tools you listed are stable or recommended by our team. Captures through the VM facilities are though, particularly VMWare.

2) Can you put the full Volatility 2 command line input/output, and also verify that you are using the latest Volatility 2 version from source?

**Comment by ikelos** [comment]
Failing all that, we can get more useful debugging information if you re-run the command but with `vol.py -vvv` rather than just `vol.py`.  It does look as though it's finding/creating the appropriate symbols which should point to the process list at the right place, but `0xc30063` seems like an odd (non-aligned) offset to be reading a page entry from, which suggests the page table automagic has detected the wrong DTB value.  The debugging information will tell us more...

**Comment by jackdyson1399** [comment]
Thank you for the comments.
I am using:
- VMware Workstation 15.5 Pro
- volatility (cloned from Git Index 703b29b)
- volatility3 (cloned from Git Index: 0d40753)

I have tried today using the .vmem file as you suggested (I included the .vmss file on the USB in the same directory that volatility is directed to as you mentioned). The issue is that I am trying to test RAM Capture tools that can be used for live scene forensics. So copying the .vmem file would rarely be an option.
The standard outputs for `imageinfo` command and the `pslist` command are shown below for the .vmem file, the DumpIT capture and the Belkasoft Capture, let me know if you need the full debugged (-vvv) outputs as they're extremely long and I felt this reply was already huge!

**Volatility:**
ImageInfo:
![image](https://user-images.githubusercontent.com/46749580/95852185-410cd380-0d4b-11eb-9e27-9ef0bcbdfd9d.png)

pslist (with Win10x64_18362 Profile):
![image](https://user-images.githubusercontent.com/46749580/95852857-63532100-0d4c-11eb-9595-0d42315d6275.png)
![image](https://user-images.githubusercontent.com/46749580/95852905-749c2d80-0d4c-11eb-92e4-946f0dec9987.png)
![image](https://user-images.githubusercontent.com/46749580/95853054-ab724380-0d4c-11eb-8cb0-1e44c2258178.png)

Interestingly, `pslist` worked fine on the DumpIT capture, but some other plugins (i.e. `consoles `, `hashdump `) failed.
![image](https://user-images.githubusercontent.com/46749580/95853317-1de32380-0d4d-11eb-9c25-cb769ac263c9.png)

**Volatility3:**
windows.info:
![image](https://user-images.githubusercontent.com/46749580/95854000-330c8200-0d4e-11eb-9160-4c1b614cd0a8.png)
![image](https://user-images.githubusercontent.com/46749580/95854027-3f90da80-0d4e-11eb-9000-e6c3100f4bee.png)

windows.pslist:
.vmem worked successfully (other plugins failed)
![image](https://user-images.githubusercontent.com/46749580/95854320-b201ba80-0d4e-11eb-8f09-91c409f62d11.png)

DumpIT started okay but failed at end due to Page error
![image](https://user-images.githubusercontent.com/46749580/95854483-eecdb180-0d4e-11eb-9577-efda9450086c.png)
...
![image](https://user-images.githubusercontent.com/46749580/95854533-00af5480-0d4f-11eb-8144-630b10923934.png)

Belkasoft started okay but failed at end due to Swap error
![image](https://user-images.githubusercontent.com/46749580/95854654-24729a80-0d4f-11eb-8953-2913f3fa8db4.png)
...
![image](https://user-images.githubusercontent.com/46749580/95854741-3fdda580-0d4f-11eb-8732-557861733915.png)

(The Magnet RAM Capture and ftk Imager captures returned the same results as Belkasoft on all tests)

p.s. sorry for the long post

**Comment by jackdyson1399** [comment]
Here are the requested files:
1.  [dumpITvol2.txt](https://github.com/volatilityfoundation/volatility3/files/5372814/dumpITvol2.txt)
2. [belkvol2.txt](https://github.com/volatilityfoundation/volatility3/files/5372812/belkvol2.txt)
3. [dumpITvol3.txt](https://github.com/volatilityfoundation/volatility3/files/5372815/dumpITvol3.txt)
4. [belkvol3.txt](https://github.com/volatilityfoundation/volatility3/files/5372864/belkvol3.txt)

Also for your interest here are the cmd outputs I received:
Vol2
![image](https://user-images.githubusercontent.com/46749580/95889182-01a9ab80-0d7a-11eb-9ffd-5277b5c2ac88.png)
Vol3
![image](https://user-images.githubusercontent.com/46749580/95889249-14bc7b80-0d7a-11eb-8f81-1808947d0e63.png)

**Comment by jackdyson1399** [comment]
Hello,
Is there any update on this please?

I have been continuing to try to get this working, in doing so I have tested some older versions of Windows 10 (1703 [15063 OS build] and 1909 [18363 OS build]. Both these VMs have had their RAM captured with the latest versions DumpIT and Belkasoft and saw the exact same issues as previously mentioned on the 2004 [19041].

I was wanting to know whether it is a Volatility Issue, a RAM Capture Tool Issue or a VMware Issue?

Also, are there any RAM Capture tools that you would recommend, as I know you previously mentioned that you do not support the four tools I am trying?

**Comment by atcuno** [comment]
I think we are close to diagnosing it. Can you please send the dumpit volatility 3 output with the full -vvv set ? Volatility 2 and 3 seem to be getting the same number of processes, but then Volatility 3 errors out at the end. The -vvv should tell us exactly where this is happening. Also, if output appears on the terimnal, and not in your redirected file, can you please send the full terminal prints as well?

**Comment by jackdyson1399** [comment]
Here's the output to the following command with -vvv (the errors only appeared in the terminal so I've included that first):
`python vol.py -f Z:\sharedCaptures\dumpPro071020.dmp -vvv windows.pslist > C:\Users\Admin\Desktop\dumpITvvv.txt`

![image](https://user-images.githubusercontent.com/46749580/96867319-752b7700-1464-11eb-9b38-24fdf161d588.png)
![image](https://user-images.githubusercontent.com/46749580/96867450-a4da7f00-1464-11eb-9081-60a4323a05ba.png)

[dumpITvvv.txt](https://github.com/volatilityfoundation/volatility3/files/5422456/dumpITvvv.txt)

**Comment by jackdyson1399** [comment]
For your interest here is a list of the commands I've tried in dumpIT and Belkasoft and their outcome.
I tried magnet and direct vmem analysis of the same machines and they had same results as Belkasoft.

The issues seem to be with reading the registry files for the cachedump, hashdump and lsadump.
[Volatility2 Testing Log.xlsx](https://github.com/volatilityfoundation/volatility3/files/5459252/Volatility2.Testing.Log.xlsx)
